service-base-quarkus/src/main/java/org/eclipse/hono/service/quarkus/AbstractServiceApplication.java

```diff
@@ -14,21 +14,32 @@
 
 package org.eclipse.hono.service.quarkus;
 
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.enterprise.event.Observes;
 import javax.inject.Inject;
 
+import org.eclipse.hono.service.ComponentNameProvider;
 import org.eclipse.hono.service.HealthCheckServer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
 import io.vertx.core.Vertx;
 import io.vertx.core.impl.cpu.CpuCoreSensor;
 
 /**
  * A base class for implementing Quarkus based services.
  *
  */
-public abstract class AbstractServiceApplication {
+public abstract class AbstractServiceApplication implements ComponentNameProvider {
```

Comment:
When testing this with the quarkus native images, the list of returned metrics is quite short (compared to the other image types):
````
# HELP hono_connections_unauthenticated  
# TYPE hono_connections_unauthenticated gauge
hono_connections_unauthenticated{component_name="hono-mqtt",component_type="adapter",host="58fe244badbc",} 0.0
# HELP jvm_threads_peak_threads The peak live thread count since the Java virtual machine started or peak was reset
# TYPE jvm_threads_peak_threads gauge
jvm_threads_peak_threads{component_name="hono-mqtt",component_type="adapter",host="58fe244badbc",} 9.0
# HELP jvm_threads_live_threads The current number of live threads including both daemon and non-daemon threads
# TYPE jvm_threads_live_threads gauge
jvm_threads_live_threads{component_name="hono-mqtt",component_type="adapter",host="58fe244badbc",} 9.0
# HELP system_cpu_count The number of processors available to the Java virtual machine
# TYPE system_cpu_count gauge
system_cpu_count{component_name="hono-mqtt",component_type="adapter",host="58fe244badbc",} 1.0
# HELP jvm_threads_daemon_threads The current number of live daemon threads
# TYPE jvm_threads_daemon_threads gauge
jvm_threads_daemon_threads{component_name="hono-mqtt",component_type="adapter",host="58fe244badbc",} 5.0
````
I guess this comes down to the others not being supported on the GraalVM (?). 

A difference to the other image types is also the content of the `host` field, here having a value of `58fe244badbc`, as opposed to something like `eclipse-hono-adapter-mqtt-vertx-79897c5d6-d92xs`. This makes it difficult to distinguish between instances in the standard `JVM details` Grafana board.
Fixing this may well be out of scope of this PR, though.

Reply:
> I guess this comes down to the others not being supported on the GraalVM (?).

yes, that is my understanding

> A difference to the other image types is also the content of the host field, here having a value of 58fe244badbc, as opposed to something like eclipse-hono-adapter-mqtt-vertx-79897c5d6-d92xs.

I didn't notice but I agree that this should be addressed outside of this issue. I am not even sure if this is an issue at all since we cannot make any assumptions about the host name anyway. That is why we have the other tags as well ...